pkg/splunk/enterprise/util.go

```diff
@@ -460,15 +460,24 @@ func ApplySmartstoreConfigMap(ctx context.Context, client splcommon.ControllerCl
 }
 
 //  setupInitContainer modifies the podTemplateSpec object
-func setupInitContainer(podTemplateSpec *corev1.PodTemplateSpec, Image string, imagePullPolicy string, commandOnContainer string) {
+func setupInitContainer(podTemplateSpec *corev1.PodTemplateSpec, Image string, imagePullPolicy string, commandOnContainer string, isEtcVolEph bool) {
+	var volMntName string
+
+	// Populate the volume mount name based on volume type(eph, pvc) and use /opt/splk/etc for init container
+	if isEtcVolEph {
+		volMntName = fmt.Sprintf(splcommon.SplunkMountNamePrefix, splcommon.SplunkMountTypeEph, splcommon.EtcVolumeStorage)
+	} else {
+		volMntName = fmt.Sprintf(splcommon.SplunkMountNamePrefix, splcommon.SplunkMountTypePvc, splcommon.EtcVolumeStorage)
+	}
```

Comment:
Can we make it a separate function, for long-term ease of maintenance?
Also, the caller can directly pass the return value of that new function, so that, it will be re-used across the places.

Reply:
Hi @sgontla , can you suggest the definition of the function? Usually the `SplunkMountNamePrefix` string is created as the volume type is known. Here its a special case where we need to pass it over and hence the logic applies only here.

Reply:
@akondur, my suggestion was to use a function, that derives the `volMntName`  based on the storage type. That way, we don't need to use the `fmt.Sprintf()` across several places. Probably, something like `getEtcMountNameByStorageType()` & `getVarMountNameByStorageType()`, which takes the storage type(ideally, an enum better serves for etc and var), and returns the mount name, that can be used across multiple containers.

Reply:
Hi @sgontla , I have created a [task](https://splunk.atlassian.net/browse/CSPL-1793) to enhance the nomenclature of the storage config code in the operator. Will be taking care of it in the next sprint as discussed.